packages/hash/api/src/model/page.model.ts

```diff
@@ -38,6 +176,123 @@ class __Page extends Entity {
     const dbEntity = await client.getEntityWithOutgoingEntityIds(params);
     return dbEntity ? new Page(dbEntity) : null;
   }
+
+  async getBlocks(client: DBClient): Promise<Block[]> {
+    const contentLinks = await this.getOutgoingLinks(client, {
+      path: ["contents"],
+    });
+
+    const blocks = await Promise.all(
+      contentLinks.map(async (link) => {
+        const destinationEntity = await link.getDestination(client);
+        return await Block.fromEntity(client, destinationEntity);
+      }),
+    );
+
+    return blocks;
+  }
+
+  async insertBlock(
+    client: DBClient,
+    params: {
+      block: Block;
+      insertedByAccountId: string;
+      position?: number;
+    },
+  ) {
+    const { position: specifiedPosition } = params;
+
+    const contentLinks = await this.getOutgoingLinks(client, {
+      path: ["contents"],
+    });
+
+    if (
+      specifiedPosition !== undefined &&
+      (specifiedPosition < 0 || specifiedPosition > contentLinks.length)
+    ) {
+      throw new UserInputError(`invalid position: ${specifiedPosition}`);
+    }
+
+    const { block, insertedByAccountId } = params;
+
+    await this.createOutgoingLink(client, {
+      destination: block,
+      stringifiedPath: "$.contents",
+      createdByAccountId: insertedByAccountId,
+      index: specifiedPosition ?? contentLinks.length,
+    });
+  }
+
+  async moveBlock(
+    client: DBClient,
+    params: {
+      currentPosition: number;
+      newPosition: number;
+      movedByAccountId: string;
+    },
+  ) {
+    const { currentPosition, newPosition } = params;
+
+    const contentLinks = await this.getOutgoingLinks(client, {
+      path: ["contents"],
+    });
+
+    // Check whether current
```

Comment:
I think this comment might have been cut off

Reply:
Good spot! I think it's fairly straightforward what's going on here so have removed the comment: https://github.com/hashintel/hash/pull/303/commits/ac5fa62e6df71cec26f8c79f3eee32f9931f5ba7